pkg/upload/koji/rh-logrus-adapter.go

```diff
@@ -1,42 +0,0 @@
-package koji
-
-import (
-	"strings"
-
-	"github.com/sirupsen/logrus"
-)
-
-type LeveledLogrus struct {
-	*logrus.Logger
-}
-
-const monitoringKeyword = "retrying"
-
-func fields(keysAndValues ...interface{}) map[string]interface{} {
-	fields := make(map[string]interface{})
-
-	for i := 0; i < len(keysAndValues)-1; i += 2 {
-		fields[keysAndValues[i].(string)] = keysAndValues[i+1]
-	}
-
-	return fields
-}
-
-func (l *LeveledLogrus) Error(msg string, keysAndValues ...interface{}) {
-	l.WithFields(fields(keysAndValues...)).Error(msg)
-}
-
-func (l *LeveledLogrus) Info(msg string, keysAndValues ...interface{}) {
-	l.WithFields(fields(keysAndValues...)).Info(msg)
-}
-func (l *LeveledLogrus) Debug(msg string, keysAndValues ...interface{}) {
-	if strings.Contains(msg, monitoringKeyword) {
-		l.WithFields(fields(keysAndValues...)).Info(msg)
-	} else {
-		l.WithFields(fields(keysAndValues...)).Debug(msg)
-	}
-}
```

Comment:
Tracing the history of this package and function brought me to this issue: https://issues.redhat.com/browse/COMPOSER-1394

We should probably keep this in mind in case the issue resurfaces now that we're simplifying the logger.  If it does, we should handle it in osbuild-composer, not here.

Reply:
We are not taking those lines out, they will be present. And since images library does not support passing context/trace_id we are not loosing anything. The caller (composer) should make sure it sets up the `olog` package writer properly and redirect logs to its (`log/slog`) logger.

Tho if we want logs to be correlated by trace_id from `slog`, that cannot be done without passing context. One idea is to copy the `slog` API by creating additional functions in the package: `PrintfContext(ctx, ...)` so external libraries could add a context hook and extract correlation id from there.

I will work on this only when I get here, there is bunch of work in composer ahead of me.

Reply:
> I will work on this only when I get here, there is bunch of work in composer ahead of me.

Yup.  Agreed.